irony.el

```diff
@@ -99,6 +99,19 @@ automatically buffer-local wherever it is set."
   "C/C++ minor mode powered by libclang."
   :group 'c)
 
+(defcustom irony-allow-inaccessibles nil
```

Comment:
This variable does not seemed used at the moment, did you forget to push some of your code?

Anyway, now I feel like it would be better to just return the availability because it might be interesting to distinguish deprecated candidate as well, and this even once the Clang bug has been resolved.

If the accessibility is exposed to Elisp, then the defcustom variable could be something like:

```el
(defcustom irony-completion-accessibility-filter '(available deprecated)
 "For completion, only accept candidates whose accessibility is in the list.

Maps to libclang's CXAvailabilityKind:
* https://clang.llvm.org/doxygen/group__CINDEX.html#ggada331ea0195e952c8f181ecf15e83d71af6542f250a1d21e256cd543669a49435

Due to a bug in Clang (BUG REFERENCE HERE), it may be interesting to enable non-accessible candidates,
blabla."
 :type '(repeat symbol)
 :options '(available deprecated not-available)
 :group 'irony-completion)
```

I think not-accessible can be left out for now and can be added later if a use if found.
By the way,  this should be moved to `irony-completion.el`.